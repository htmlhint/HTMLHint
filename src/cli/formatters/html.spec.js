const ChildProcess = require('child_process')
const path = require('path')
const serializer = require('jest-serializer-path')

expect.addSnapshotSerializer(serializer)

describe('CLI', () => {
  describe('Formatter: html', () => {
    it('should have stdout output with formatter html', (done) => {
      ChildProcess.exec(
        [
          'node',
          path.resolve(__dirname, '../../../bin/htmlhint'),
          path.resolve(__dirname, '__fixtures__', 'example.html'),
          '--format',
          'html',
        ].join(' '),
        (error, stdout, stderr) => {
          expect(typeof error).toBe('object')
          expect(error.code).toBe(1)

          expect(stdout).toMatchSnapshot()

          expect(stderr).toBe('')
          done()
        }
      )
    })
  })
})
