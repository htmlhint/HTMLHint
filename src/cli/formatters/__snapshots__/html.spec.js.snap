// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Formatter: html should have stdout output with formatter html 1`] = `
"<html><head><title>HTML Hint Violation Report</title></head><body><center><h2>Violation Report</h2></center><table border=\\"1\\"><tr><th>Number#</th><th>File Name</th><th>Line Number</th><th>Message</th></tr><tr><td>1</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>8</td><td>The value of attribute [ bad ] must be in double quotes.</td></tr><tr><td>2</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>8</td><td>The value of attribute [ bad ] must be in double quotes.</td></tr><tr><td>3</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>8</td><td>Duplicate of attribute name [ bad ] was found.</td></tr><tr><td>4</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>9</td><td>The value of attribute [ bad ] must be in double quotes.</td></tr><tr><td>5</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>9</td><td>The value of attribute [ bad ] must be in double quotes.</td></tr><tr><td>6</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>9</td><td>Duplicate of attribute name [ bad ] was found.</td></tr><tr><td>7</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>10</td><td>Tag must be paired, no start tag: [ </input> ]</td></tr><tr><td>8</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>11</td><td>Special characters must be escaped : [ < ].</td></tr><tr><td>9</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>11</td><td>Special characters must be escaped : [ > ].</td></tr><tr><td>10</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>13</td><td>Tag must be paired, no start tag: [ </div> ]</td></tr><tr><td>11</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>14</td><td>Tag must be paired, no start tag: [ </div> ]</td></tr><tr><td>12</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>15</td><td>Tag must be paired, no start tag: [ </hello> ]</td></tr><tr><td>13</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>16</td><td>Tag must be paired, no start tag: [ </test> ]</td></tr><tr><td>14</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>17</td><td>Tag must be paired, no start tag: [ </div> ]</td></tr><tr><td>15</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>21</td><td>The value of attribute [ class ] must be in double quotes.</td></tr><tr><td>16</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>21</td><td>The value of attribute [ what ] must be in double quotes.</td></tr><tr><td>17</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>21</td><td>The value of attribute [ something ] must be in double quotes.</td></tr><tr><td>18</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>25</td><td>Tag must be paired, no start tag: [ </div> ]</td></tr><tr><td>19</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>26</td><td>Tag must be paired, no start tag: [ </bad> ]</td></tr><tr><td>20</td><td><PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html</td><td>27</td><td>Tag must be paired, no start tag: [ </bad> ]</td></tr></table></body></html>
"
`;
